engines/stark/metaengine.cpp

```diff
@@ -171,6 +177,147 @@ class StarkMetaEngine : public AdvancedMetaEngine {
+	act->addDefaultInputMapping("F1");
+	gameKeyMap->addAction(act);
+	
+	act = new Action("SAVEGAME", _("Save Game"));
+	act->setCustomEngineActionEvent(kActionSaveGame);
+	act->addDefaultInputMapping("F2");
+	gameKeyMap->addAction(act);
+	
+	act = new Action("LOADGAME", _("Load Game"));
+	act->setCustomEngineActionEvent(kActionLoadGame);
+	act->addDefaultInputMapping("F3");
+	gameKeyMap->addAction(act);
+	
+	act = new Action("CONVOLOG", _("Conversation log"));
+	act->setCustomEngineActionEvent(kActionConversationLog);
+	act->addDefaultInputMapping("F4");
+	gameKeyMap->addAction(act);
+	
+	act = new Action("APRILSDIARY", _("April's Diary (initially disabled)"));
+	act->setCustomEngineActionEvent(kActionAprilsDiary);
+	act->addDefaultInputMapping("F5");
+	gameKeyMap->addAction(act);
+	
+	act = new Action("VIDREPLAY", _("Video Replay"));
+	act->setCustomEngineActionEvent(kActionVideoReplay);
+	act->addDefaultInputMapping("F6");
+	gameKeyMap->addAction(act);
+	
+	act = new Action("GAMESETTINGS", _("Game Settings"));
+	act->setCustomEngineActionEvent(kActionGameSettings);
+	act->addDefaultInputMapping("F7");
+	gameKeyMap->addAction(act);
+	
+	act = new Action("SAVESCRNSHOT", _("Save Screenshot"));
+	act->setCustomEngineActionEvent(kActionSaveScreenshot);
+	act->addDefaultInputMapping("F8");
+	gameKeyMap->addAction(act);
+	
+	act = new Action("TOGGLESUBS", _("Toggle Subtitles"));
+	act->setCustomEngineActionEvent(kActionToggleSubtitles);
+	act->addDefaultInputMapping("F9");
+	gameKeyMap->addAction(act);
+	
+	act = new Action("QUITTOMENU", _("Quit to Menu"));
+	act->setCustomEngineActionEvent(kActionQuitToMenu);
+	act->addDefaultInputMapping("F10");
+	gameKeyMap->addAction(act);
+	
+	act = new Action("CYCLEBACK", _("Cycle back through inventory cursor items"));
+	act->setCustomEngineActionEvent(kActionCycleForwardInventory);
+	act->addDefaultInputMapping("a");
+	gameKeyMap->addAction(act);
+	
+	act = new Action("CYCLEFORWARD", _("Cycle forward through inventory cursor items"));
+	act->setCustomEngineActionEvent(kActionCycleBackInventory);
+	act->addDefaultInputMapping("s");
+	gameKeyMap->addAction(act);
+	
+	act = new Action("INVENTORY", _("Inventory"));
+	act->setCustomEngineActionEvent(kActionInventory);
+	act->addDefaultInputMapping("i");
+	gameKeyMap->addAction(act);
+	
+	act = new Action("DISPLAYEXITS", _("Display all exits on current location"));
+	act->setCustomEngineActionEvent(kActionDisplayExits);
+	act->addDefaultInputMapping("x");
+	gameKeyMap->addAction(act);
+
+	act = new Action("EXITGAME", _("Exit Game"));
+	act->setCustomEngineActionEvent(kActionExitGame);
+	act->addDefaultInputMapping("A+x");
+	act->addDefaultInputMapping("A+q");
+	gameKeyMap->addAction(act);
+
+	act = new Action("PAUSE", _("Pause Game"));
+	act->setCustomEngineActionEvent(kActionPause);
+	act->addDefaultInputMapping("p");
+	gameKeyMap->addAction(act);
+
+	act = new Action("SCROLLUP", _("Scroll up in inventory or in your dialogues"));
```

Comment:
This makes me wonder if there should be a separate keymap for when the inventory is open.

It might be useful to have joystick bindings for these as well.

Reply:
need some guidance on how to use a different keymap or why would it be necessary here?

Reply:
I think this could be to have separate inventory scroll up / down actions, dialogue scroll up / down actions, and dialogue prev / next actions. These could have default maps to all the current values and simply generate both actions when matched by a shared input.
The dialogue handlers would likely need to check !isInventoryOpen() if this were the case.

@ccawley2011 - Is this what you were considering?